### DISCUSSION TITLE
Webpack assignment error: Cannot read property 'length' of undefined

### DISCUSSION THREAD

**Discussion by nashvegasnate** [nss-evening-cohort-14/evening-client-side#112] [Help Me!] [answered]
I solved all the errors in the terminal on this assignment, however, once I try and run the app and enter a name and submit the form, I get this error:
Uncaught TypeError: Cannot read property 'length' of undefined
    at cardCreator (main.js?27fa:38)
    at HTMLDivElement.addStudentToArray (main.js?27fa:76)

Three search URLs:
https://trackjs.com/blog/debugging-cannot-read-property-length-of-undefined/#:~:text=This%20error%20indicates%20that%20our,error%2C%20and%20execution%20will%20stop.

https://medium.com/@gokulraj112/typeerror-cannot-read-property-length-of-undefined-javascript-71120097ee6#:~:text=As%20a%20Result%20you%20have,as%20to%20avoid%20these%20errors.

https://www.codecademy.com/forum_questions/5385f1c09c4e9db84f002028

What I have tried:
I tried to convert the cardCreator for loop into a for each loop but couldn't seem to get it to work.

Link to repo:
https://github.com/nashvegasnate/sorting-hat-webpack

**Comment by TrinityChristiana** [comment, marked as answer]
The [sortStudentsByHouse function](https://github.com/nashvegasnate/sorting-hat-webpack/blob/bd2f080446178b0f1ec93c5026e15ae9701d9f3d/src/javascripts/main.js#L32-L34) is missing a return statement.

**Reply by nashvegasnate** [reply]
So, when I was clearing up all the errors in the code as per the assignment, I had initially gotten this error in my terminal:
  32:40  error  Unexpected block statement surrounding arrow body; move the returned value immediately after the `=>`  arrow-body-style
So, when I googled it, the solution was telling me that I didn't need to include it, so I removed it and my code turned green. However, on the served page, I got the aforementioned error listed above.

**Reply by TrinityChristiana** [reply]
When you get that error, change the curly brackets of the function throwing that error to parenthesis and remove the return statement. 

So this function...
```js
const sortStudentsByHouse = (array) => {
  return array.sort((a, b) => (a.house > b.house ? 1 : -1));
};
```
...would look like this:
```js
const sortStudentsByHouse = (array) => ( 
  array.sort((a, b) => (a.house > b.house ? 1 : -1))
);
```
or
```js
const sortStudentsByHouse = (array) =>  array.sort((a, b) => (a.house > b.house ? 1 : -1))
```

**Reply by nashvegasnate** [reply]
Woohoo! That worked! I figured it was a simple solution, especially since I've had some trouble wrapping my head around JS. Thank you for that help.

**Reply by TrinityChristiana** [reply]
Glad we could help!
